### PR TITLE
System lock formula

### PR DESCRIPTION
**Pull request by mateiw**
Formula to control the Salt blackout mode.

Screen Shot 2019-12-17 at 16 25 17

### CODE REVIEW

File: salt-blackout-formula/metadata/form.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# This file is part of blackout-formula.
+#
+# Foobar is free software: you can redistribute it and/or modify
```
Comment (mcalmer) [inline salt-blackout-formula/metadata/form.yml]: Too many foobar in the headers. Here and in the other files.

Comment (mateiw) [reply salt-blackout-formula/metadata/form.yml]: Thanks. Removed headers.

File: salt-blackout-formula/metadata/form.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  $help: Modules allowed during blackout. Removing the default values is not recommended because it can prevent Uyuni from working properly.
+  $disabled: "!formValues.minion_blackout"
+  $prototype:
+    $type: text
+  $default:
+  - test.ping
+  - grains.item
+  - grains.items
+  - mgractionchains.start
+  - mgractionchains.start
```
Comment (aaannz) [inline salt-blackout-formula/metadata/form.yml]: Why is the `mgractionchains.start` listed twice?

File: salt-blackout-formula/metadata/pillar.example [hunk 3]
```diff
@@ -0,0 +1,10 @@
+minion_blackout: true
+minion_blackout_whitelist:
+- test.ping
+- grains.item
+- grains.items
+- mgractionchains.start
+- mgractionchains.start
```
Comment (aaannz) [inline salt-blackout-formula/metadata/pillar.example]: Same as above, listed twice.

File: salt-blackout-formula/salt-blackout-formula.changes [hunk 4]
```diff
@@ -0,0 +1,4 @@
+-------------------------------------------------------------------
+Tue Oct 15 16:41:21 UTC 2019 - malbu@suse.com
+
+- Initial version of prometheus-formula
```
Comment (aaannz) [inline salt-blackout-formula/salt-blackout-formula.changes]: Copy paste from the other formula

File: salt-blackout-formula/salt-blackout-formula.spec [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+# case the license is the MIT License). An "Open Source License" is a
+# license that conforms to the Open Source Definition (Version 1.9)
+# published by the Open Source Initiative.
+
+# Please submit bugfixes or comments via https://bugs.opensuse.org/
+#
+
+
+%define fname salt-blackout
+%define fdir  %{_datadir}/susemanager/formulas
```
Comment (aaannz) [inline salt-blackout-formula/salt-blackout-formula.spec]: Maybe we can start using the new `/usr/share/salt-formulas` directory as requested in https://github.com/SUSE/spacewalk/issues/10230

File: salt-blackout-formula/metadata/metadata.yml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+description:
+  Change minion blackout mode. When in blackout mode, all remote execution commands are disabled.
+group: general_system_configuration
+after:
+  - users
```
Comment (aaannz) [inline salt-blackout-formula/metadata/metadata.yml]: The after probably isn't needed, or is there a reason for the ordering?

File: salt-blackout-formula/metadata/form.yml (hunk 2 above)
Comment (mateiw) [reply salt-blackout-formula/metadata/form.yml]: Removed `mgractionchains` completely. Not needed in the first iteration.

**Pull request merged**